CPU Cumulative fix

awilliamson opened the pull request:
Issue identified by @Jazzelhawk. This was fixed on awilliamson@e432737094f37a26aee4b6a6a633e6b69fd7af0e by implementation of a disposition system. However, this system was untested and may have been too much of a change from the norm.

This cumulative fix is long overdue, and may require tweaking on `sf_timebuffer` lower and `sf_timebuffersize` higher.

If somebody would like to suggest new values, I recommend ~0.003 and 64 respectively. These aren't included in this PR as they are speculative and dependent upon each server, but should be investigated prior to acceptance.

awilliamson committed:
Fixed cumulative error in cpuTime

awilliamson committed:
Corrected CPU Exceeded message.

Jazzelhawk commented:
Now that the `oldSysTime` variable has been fixed we can't do `self.cpuTime.current = SysTime() - oldSysTime` anymore since this is just the time passed since the last cpuCheck. It should be set the same value as `self.cpuTime.buffer[ ind ]`.

awilliamson commented:
`self.cpuTime.current` acts as a delta between 500 LuaC executes. It is then incremented onto the current buffer by `self.cpuTime.buffer[ ind ]`.

awilliamson committed:
Fixed quotaUsed to use current buffer.

Was using cpuTime.current, which was fixed to delta-time. Therefore,
this needed updating to use the current bin, something which
cpuTime.current was previously intended.

awilliamson committed:
Localised cpuTime.current as dt. Removed refs

Changed cpuTime.current to be local to the cpu check, as dt. Removed
initialisation of current member table of cpuTime, and any zeroing
steps done elsewhere.
